Update makefile to add common dev tasks using opa docker image.

gkowalski-google opened the pull request:
Resolves #318

morgante commented on Makefile:
```diff
@@ -15,13 +15,24 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
+OPA_IMAGE := opa
```
Should we use the image we maintain? https://github.com/forseti-security/policy-library/blob/master/cloudbuild.yaml

gkowalski-google replied:
Let me try using that one. With the official OPA image, `fmt -d` does not work because `diff` cannot be found. Maybe it will work with this image.

gkowalski-google replied:
I'm not running it right.
```
docker run -it --rm \
                -v $(CURDIR):/workspace \
                gcr.io/config-validator/rego-v0.16.0:latest \
                test
```
produces error:
```
make: *** No rule to make target 'test'.  Stop.
make: *** [docker_test] Error 2
```

morgante replied:
This command worked for me, you need to set the working directory (or we could update the Dockerfile to include WORKDIR):
```
docker run -it --rm \
  -v "$(pwd):/workspace" \
  -w=/workspace \
  gcr.io/config-validator/rego-v0.16.0:latest \
  test
```

gkowalski-google replied:
Nice, thanks! This works for me now.

briantkennedy commented on Makefile:
```diff
@@ -15,13 +15,24 @@
 
+.PHONY: docker_test
+docker_test: ## Run tests using OPA docker image
+	docker run -it --rm \
```
can you adjust this to run tests using opa 0.15.x and .016.2

gkowalski-google replied:
Updated using OPA image to run tests on 0.15.1 + 0.16.2 by running `make docker_test`.

briantkennedy commented on Makefile:
```diff
@@ -45,6 +56,13 @@ check_sample_files: ## Make sure each template in policies/templates has one sam
 
+.PHONY: docker_check_format
+docker_check_format: ## Check format of rego using OPA docker image
+	docker run -it --rm \
```
Let's keep this consistent with the cloudbuild and use docker 0.15.x.

gkowalski-google replied:
Updated.

gkowalski-google commented:
> Alternatively, we could look at using the opa image in cloud build and using cloud-build-local to execute things at pre-submit time

@morgante @briantkennedy Let me know what your preference is. I was using the OPA image because I couldn't get the CV image working, but Morgante helped with that. I'm happy to make the changes. The motivation for me was to make it easy to run the tests on mac, where I frequently run into issues with certain software blocking me from running the OPA executable.

morgante commented:
> @morgante @briantkennedy Let me know what your preference is. I was using the OPA image because I couldn't get the CV image working, but Morgante helped with that. I'm happy to make the changes. The motivation for me was to make it easy to run the tests on mac, where I frequently run into issues with certain software blocking me from running the OPA executable.

I think we should use the same image in both. It'd probably be easiest for you to update the Makefile to use the CV image.

gkowalski-google commented:
I've updated to run the tests using the CV rego image. Also added in targets to run the lint tests, check sample files, and check the formatting. Everything working for me locally.
